File: solutions/python/prime-factors/1/prime_factors.py
primes = [2, 3]


def factors(value: int) -> list[int]:
    result = []
    for p in primes:
        if p > value:
            break
        while value % p == 0:
            result.append(p)
            value //= p
    x = primes[-1] + 2
    while value != 1:
        if all(x % p for p in primes):
            while value % x == 0:
                result.append(x)
                value //= x
        x += 2
    return result

File: solutions/go/prime-factors/1/prime_factors.go
package prime

var primes = []int64{2, 3}

func Factors(n int64) []int64 {
	r := make([]int64, 0)
	for _, p := range primes {
		if p > n {
			break
		}
		for n%p == 0 {
			r = append(r, p)
			n /= p
		}
	}
	for x := primes[len(primes)-1] + 2; n != 1; x += 2 {
		xIsPrime := true
		for _, p := range primes {
			if x%p == 0 {
				xIsPrime = false
				break
			}
		}
		if xIsPrime {
			primes = append(primes, x)
			for n%x == 0 {
				r = append(r, x)
				n /= x
			}
		}
	}
	return r
}
